packages/react-components/react-drawer/src/components/Drawer/useDrawer.ts

```diff
@@ -73,12 +46,19 @@ export const useDrawer_unstable = (props: DrawerProps, ref: React.Ref<HTMLElemen
     initialState: false,
   });
 
-  const { dialog, dialogSurface } = useDrawerDialogProps({
-    ...props,
-    lightDismiss,
-    open,
-    modal,
-  });
+  const dialogProps = React.useMemo(() => {
+    return {
+      modalType: getModalType(modal, lightDismiss),
+      open,
+      onOpenChange,
+      children,
+    } as DialogProps;
+  }, [children, lightDismiss, modal, onOpenChange, open]);
+
+  const dialogSurfaceProps = {
```

Comment:
How will users override styles for the `backdrop` slot in `DialogSurface`

Reply:
It makes sense. But as this isn't part of this PR's scope, it'll be fixed on another PR. The work can be tracked here:
https://github.com/marcosmoura/fluentui/tree/feat/split-drawer/packages/react-components/react-drawer/src